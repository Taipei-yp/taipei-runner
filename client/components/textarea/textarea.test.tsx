import React from "react";
import TestRenderer, { act } from "react-test-renderer";
import { Textarea } from ".";

describe("Component Text", () => {
  let onChangeCall = false;
  const changeFunc = () => {
    onChangeCall = true;
  };
  const textarea = TestRenderer.create(
    <Textarea
      value="testvalue"
      name="testname"
      className="testclass"
      onChange={changeFunc}
    />,
  );

  it("correct render", () => {
    expect(textarea.toJSON()).toMatchSnapshot();
  });
  it("correct props", () => {
    expect(textarea.root.props.name).toBe("testname");
    expect(textarea.root.props.className).toBe("testclass");
  });
  it("onchange work", () => {
    act(textarea.root.props.onChange);
    expect(onChangeCall).toBe(true);
  });
});
