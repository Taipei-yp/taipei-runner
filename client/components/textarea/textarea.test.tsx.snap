// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Textarea correct render 1`] = `
<textarea
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
  name="testname"
  onChange={[MockFunction]}
  value="testvalue"
/>
`;
